File: LeetCode/MissAndRepeat.java
package LeetCode;


public class MissAndRepeat {
    public static void main(String[] args) {
        int[] a = {1,1,2,3,4};
        int i;

        int[] op = new int[a.length];
        for(i=0;i<a.length;i++){
            if(op[a[i]-1] != 0 ) 
                System.out.println("Repeat "+a[i]);
            op[a[i]-1] =  1;
        }
        for(i=0;i<a.length;i++)
            if(op[i] == 0)
                System.out.println("Miss "+(i+1));
    }
}
//    X X X
//    X 0 X
//    X X X

File: LeetCode/MissAndRepeat.js
const main = () => {
  // const a = [2, 2, 3, 4, 5];
  const a = [1, 1, 2, 3, 4];
  let k = a[0],
    miss = -1,
    repeat = -1;
  for (let i = 1; i < a.length - 1; i++) {
    ++k;
    if (a[i] === a[i - 1]) repeat = a[i];
    else if (a[i] != k && a[i + 1] != k) miss = a[i];
  }
  console.log(`repeat - ${repeat} \n miss - ${miss === -1 ? k + 1 : miss}`);
};
main();
// Repeat 1
// Miss 5
